### PR TITLE
Arm Baremetal Runner cleanup

### PR DESCRIPTION
**Pull request by robell**
This converts the arm_executor_runner into a examples/arm/ standalone cmake project which depends on the Ethos-U sdk and the Executorch runtime for bare metal, rather than being a series of patches.

This cleans up all but a few issues, most prominent remaining are:
 * removing the final patch against the sdk linker script (WIP)
 * linking against the CMAKE executorch project cleanly rather than importing individual libraries (there isn't currently a CMAKE target for third party projects outside executorch repo)

The next anticipated changes would be to remove the remaining linker script patch and vela patch so we are working with clean checkouts of the respective projects.

### CODE REVIEW

File: examples/arm/executor_runner/arm_executor_runner.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    ET_LOG(
+        Info,
+        "Loading of method %s failed with status 0x%" PRIx32,
+        method_name,
+        method.error());
+  }
+  ET_LOG(Info, "Method loaded.");
+
+  ET_LOG(Info, "Preparing inputs...");
+  auto inputs = torch::executor::util::prepare_input_tensors(*method);
```
Comment (larryliu0820) [inline examples/arm/executor_runner/arm_executor_runner.cpp:129]: Can we check if inputs is ok here

**Review by larryliu0820 — APPROVED** [review]
Can you rebase and see if `test-arm-backend-delegation` is passing? You can trigger it by adding `ciflow/trunk` tag to this PR.

File: examples/arm/executor_runner/arm_executor_runner.cpp (hunk 1 above)
Comment (robell) [reply examples/arm/executor_runner/arm_executor_runner.cpp:129]: done.

**Comment by robell** [comment]
reference-delegation still fails due to #1947 as a seperate problem, so this seems ok to merge and the 0010- patch has been resolved into the new baremetal_runner.

**Comment by digantdesai** [comment]
> Can you rebase and see if `test-arm-backend-delegation` is passing? You can trigger it by adding `ciflow/trunk` tag to this PR.

Curious - can we do it on `partner: arm` automagically?

**Review by digantdesai — APPROVED** [review]
Looks good to me. Let me know when you think this is ready for merging.

**Comment by robell** [comment]
@digantdesai I think this is ready - sooner we can merge the better!

**Pull request closed without merging**